arelle/plugin/OimTaxonomy/XbrlConcept.py

```diff
@@ -0,0 +1,47 @@
+"""
+See COPYRIGHT.md for copyright information.
+"""
+
+from typing import TYPE_CHECKING, TypeAlias, Optional
+from decimal import Decimal
+
+from arelle.ModelValue import QName
+from arelle.PythonUtil import OrderedSet
+from .XbrlProperty import XbrlProperty
+from .XbrlTypes import XbrlTaxonomyType, QNameKeyType
+from .XbrlTaxonomyObject import XbrlTaxonomyObject
+
+XbrlUnitTypeType: TypeAlias = "XbrlUnitType"
+
+class XbrlConcept(XbrlTaxonomyObject):
+    taxonomy: XbrlTaxonomyType
+    name: QNameKeyType # (required) The name is a QName that uniquely identifies the concept object.
+    dataType: QName # (required) Indicates the dataType of the concept. These are provided as a QName based on the datatypes specified in the XBRL 2.1 specification and any custom datatype defined in the taxonomy.
+    periodType: QName # (required) Indicates the period type of the concept. The property values can be either instant or duration. If the concept can be an atemporal value it must be defined as a duration. (i.e. the value does not change with the passage of time)
```

Comment:
Since this is likely the beginning of the new data model for Arelle well into the future, I'd like to make an effort toward conforming to conventional Python styling. Per [PEP 8](https://peps.python.org/pep-0008/#function-and-variable-names), these property names should all be snake_case. 
Patch based on a quick regex find-and-replace: [patch.diff.txt](https://github.com/user-attachments/files/19805988/patch.diff.txt)